packages/@uppy/react/src/DashboardModal.js

```diff
@@ -31,6 +31,12 @@ class DashboardModal extends React.Component {
     } else if (!prevProps.open && this.props.open) {
       this.plugin.openModal()
     }
+
+    for (const key of Object.keys(this.props)) {
+      if (this.props[key] !== prevProps[key] && key in this.plugin.opts) {
+        this.plugin[key] = this.props[key]
+      }
+    }
```

Comment:
Alright. But we some suggestions:

- should use  `.setOptions` for to do this.
- optional: simplify with some equality util or micro lib, like we do in Vue.

```suggestion
    if (!shallowEqualObjects(this.props, prevProps)) {
      this.plugin.setOptions(this.props)
    }
```